tests/update_workflow_test.go

```diff
@@ -1326,16 +1326,21 @@ func (s *UpdateWorkflowSuite) TestUpdateWorkflow_ValidateWorkerMessages() {
 			_, err := poller.PollAndProcessWorkflowTask()
 			updateResult := <-updateResultCh
 			if tc.RespondWorkflowTaskError != "" {
-				require.Error(s.T(), err, "RespondWorkflowTaskCompleted should return an error contains `%v`", tc.RespondWorkflowTaskError)
-				require.Contains(s.T(), err.Error(), tc.RespondWorkflowTaskError)
-
-				// When worker returns validation error, API caller got timeout error.
-				require.Error(s.T(), updateResult.err)
-				require.True(s.T(), common.IsContextDeadlineExceededErr(updateResult.err), updateResult.err.Error())
-				require.Nil(s.T(), updateResult.response)
+				s.ErrorContainsf(err, tc.RespondWorkflowTaskError,
+					"RespondWorkflowTaskCompleted should return an error containing `%v`", tc.RespondWorkflowTaskError)
+
+				if updateResult.err != nil {
+					// Usually, the client times out the request before the server can respond.
+					s.Error(updateResult.err)
+					s.True(common.IsContextDeadlineExceededErr(updateResult.err), updateResult.err.Error())
+					s.Nil(updateResult.response)
+				} else {
+					// This happens when the server responds before the client times out the request.
+					s.Nil(updateResult.response.GetStage(), enumspb.UPDATE_WORKFLOW_EXECUTION_LIFECYCLE_STAGE_ADMITTED)
```

Comment:
My hypothesis is that _occasionally_  the server somehow (?) manages to respond before the client realized that the time is up. Otherwise I don't know how to explain that this sometimes occurs.